### PR TITLE
Deprecate vttablet metrics `QueryCacheXX` and rename to `TabletQueryPlanCacheXX`

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR is a follow up of https://github.com/vitessio/vitess/pull/14947, where we added two new metrics but in the end we would like to rename to `TabletQueryPlanCache` for more clarity and to avoid the ambiguity with an actual query cache.

All the `QueryCacheXX` metrics in VTTablet are renamed.

We will need to update the release notes of `v20.0.0` and backport only the release notes change to `release-20.0`, I am adding the backport to label, but the backport should not be merge without changing its content.

#### Before
image
image

#### After
image
image

### CODE REVIEW

File: config/tablet/default.yaml [hunk 1]
```diff
@@ -99,7 +99,6 @@ hotRowProtection:
 consolidator: enable|disable|notOnPrimary # enable-consolidator, enable-consolidator-replicas
 passthroughDML: false                    # queryserver-config-passthrough-dmls
 streamBufferSize: 32768                  # queryserver-config-stream-buffer-size
-queryCacheSize: 5000                     # queryserver-config-query-cache-size
```
Comment (frouioui) [inline config/tablet/default.yaml:102]: This flag was removed a while ago, I took the liberty of removing this from the config file.

File: changelog/20.0/20.0.0/release_notes.md [hunk 2]
```diff
@@ -360,8 +360,12 @@ You can now run `VDiff`s on OnlineDDL schema change migrations, which are not ye
...
 VTTablet exposes two new counter stats:
 
- * `QueryCacheHits`: Query engine query cache hits
- * `QueryCacheMisses`: Query engine query cache misses
+ * `QueryCacheHits`: Query engine query plan cache hits
+ * `QueryCacheMisses`: Query engine query plan cache misses
+
+> [!NOTE]
+> `QueryCache` does not refer to a query cache, but to a query plan cache.
+> In v21, these metrics will be deprecated and replaced by `TabletQueryPlanCacheHits` and `TabletQueryPlanCacheMisses`.
```
Comment (deepthi) [inline changelog/20.0/20.0.0/release_notes.md:368]: The reason metric names are not prefixed with `Tablet` or `Gate` is that the binary name is automatically added to the metric name as a prefix. So this becomes `vttablet_query_plan_cache_misses` if you look at `/metrics`.
All of the metrics introduced in this PR need to be renamed and the `Tablet` prefix should be removed.

**Review by deepthi — COMMENTED** [review]
Can you please address the naming issue?
For metrics PRs, we should always provide a screenshot showing the changed section from the output of the `/metrics` endpoint and `/debug/vars` endpoint. Looking at `/metrics` would have immediately shown you what I said in my review comment.

File: changelog/20.0/20.0.0/release_notes.md (hunk 2 above)
Comment (frouioui) [reply changelog/20.0/20.0.0/release_notes.md:368]: That won't work as the vtgate and vttablet metrics will have the same name in vtcombo, leading to a panic/failure.

Comment (frouioui) [reply changelog/20.0/20.0.0/release_notes.md:368]: I will have to see how I can tweak vtcombo

**Comment by deepthi** [comment]
Why not do a separate PR just for the v20 release notes? then this can take as much time as necessary.

**Comment by frouioui** [comment]
> Why not do a separate PR just for the v20 release notes? then this can take as much time as necessary.

That's a good point: https://github.com/vitessio/vitess/pull/16299

**Review by harshit-gangal — APPROVED** [review]

File: go/vt/vttablet/tabletserver/query_engine_test.go [hunk 3]
```diff
@@ -197,26 +197,26 @@ func TestQueryPlanCache(t *testing.T) {
 	ctx := context.Background()
 	logStats := tabletenv.NewLogStats(ctx, "GetPlanStats")
 
-	initialHits := qe.queryCacheHits.Get()
-	initialMisses := qe.queryCacheMisses.Get()
```
Comment (deepthi) [inline go/vt/vttablet/tabletserver/query_engine_test.go:201]: In order to ensure that we keep testing the old metrics until we delete them, we should keep the tests for them. That will require keeping the struct members also around until the metrics are deleted. Each vttablet has one instance of the queryEngine, so it should not be a big deal.

**Review by deepthi — COMMENTED** [review]
We should do a similar PR for vtgate after this is complete.

File: go/vt/vttablet/tabletserver/query_engine_test.go (hunk 3 above)
Comment (frouioui) [reply go/vt/vttablet/tabletserver/query_engine_test.go:201]: Done via ade9210

**Comment by frouioui** [comment]
> We should do a similar PR for vtgate after this is complete.

VTGate is already using the proper name. See https://github.com/vitessio/vitess/blob/75aee8ab0c842d296c32d6b2cd1b06bdb84e6788/go/vt/vtgate/executor.go#L195-L212

**Review by deepthi — APPROVED** [review]

**Pull request merged**